framework/src/main/java/org/checkerframework/common/reflection/ClassValAnnotatedTypeFactory.java

```diff
@@ -287,7 +287,22 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
                     ClassTree classTree = TreePathUtil.enclosingClass(getPath(tree));
                     clType = (Type) TreeUtils.typeOf(classTree);
                 }
-                String className = getClassNameFromType(clType);
+                String className;
+                if (clType.tsym.isAnonymous()) {
```

Comment:
https://github.com/SirYwell/checker-framework/blob/0281206f6fce7410b4908ea25741c8fd89114186/framework/src/main/java/org/checkerframework/common/reflection/ClassValAnnotatedTypeFactory.java#L347 says "This will not return the correct name for anonymous classes."
Wouldn't it be better if this were fixed in `getClassNameFromType`?

That method is also called [in MemberSelect](https://github.com/SirYwell/checker-framework/blob/0281206f6fce7410b4908ea25741c8fd89114186/framework/src/main/java/org/checkerframework/common/reflection/ClassValAnnotatedTypeFactory.java#L261). Can you add a test with a member select on an anonymous type?

Reply:
You can't really have a class literal expression for an anonymous type. That's also one reason why I decided to handle it outside the `getClassNameFromType` method.

The other reason was because I initially produced multiple bounds, one for the super class and (potentially) one for the super interface. I rewrote that part because it doesn't make much sense, so now that we're back at one class name, there is nothing preventing us from moving the code to `getClassNameFromType`.